File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/Domain/PEC/PECMailBox.vb
<Serializable()> _
Public Class PECMailBox
    Inherits DomainObject(Of Short)

#Region " Constructor "

    Public Sub New()
        UniqueId = Guid.NewGuid()
    End Sub


#End Region

#Region " Properties "

    Public Overridable Property MailBoxName As String

    Public Overridable Property IncomingServerName As String

    Public Overridable Property IncomingServerProtocol As IncomingProtocol?

    Public Overridable Property IncomingServerPort As Integer?

    Public Overridable Property IncomingServerUseSsl As Boolean?

    Public Overridable Property OutgoingServerName As String

    Public Overridable Property OutgoingServerPort As Integer?

    Public Overridable Property OutgoingServerUseSsl As Boolean?

    Public Overridable Property Username As String

    Public Overridable Property Password As String

    Public Overridable Property Mails As IList(Of PECMail)

    Public Overridable Property Roles As IList(Of Role)

    ''' <summary>Collegamento tra caselle di posta e settori</summary>
    ''' <remarks>Aggiunto per poter accedere alla proprietà decorata <see>PECMailBoxRole.Priority</see></remarks>
    Public Overridable Property MailBoxRoles As IList(Of PECMailBoxRole)

    Public Overridable Property Managed As Boolean

    Public Overridable Property Unmanaged As Boolean

    Public Overridable Property Location As Location

    Public Overridable Property IsDestinationEnabled As Nullable(Of Boolean)

    Public Overridable Property IsForInterop As Boolean

    Public Overridable Property Configuration As PECMailBoxConfiguration

    Public Overridable Property IsHandleEnabled As Boolean?

    Public Overridable Property IsProtocolBox As Boolean?

    Public Overridable Property IsProtocolBoxExplicit As Boolean?

    Public Overridable Property IdJeepServiceIncomingHost As Guid?

    Public Overridable Property IdJeepServiceOutgoingHost As Guid?

    Public Overridable Property RulesetDefinition As String

    Public Overridable Property InvoiceType As InvoiceType?

    Public Overridable Property HumanEnabled As Boolean

#End Region

#Region " Methods "

    ''' <summary>
    ''' Indica se la mailbox corrente ha sia il server in ingresso che in uscita configurati
    ''' </summary>
    ''' <remarks>
    ''' Indica se GESTITA / NON GESTITA, soluzione temporanea perchè le colonne MANAGED e UNMANAGED sono usate e non è disponibile un'analisi.
    ''' </remarks>
    Public Overridable Function HasServers() As Boolean
        Return Not String.IsNullOrEmpty(OutgoingServerName) AndAlso Not String.IsNullOrEmpty(IncomingServerName)
    End Function

    Public Overridable Function IsSendingEnabled() As Boolean
        Return Not String.IsNullOrEmpty(OutgoingServerName)
    End Function

#End Region

End Class

File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/PECMails/PECMailBox.cs
using System;
using System.Collections.Generic;
using VecompSoftware.DocSuiteWeb.Entity.Commons;
using VecompSoftware.DocSuiteWeb.Entity.OCharts;
using VecompSoftware.DocSuiteWeb.Entity.Tenants;

namespace VecompSoftware.DocSuiteWeb.Entity.PECMails
{

    public class PECMailBox : DSWBaseEntity
    {
        #region [ Constructor ]

        public PECMailBox() : this(Guid.NewGuid()) { }

        public PECMailBox(Guid uniqueId)
            : base(uniqueId)
        {
            PECMails = new HashSet<PECMail>();
            OChartItems = new HashSet<OChartItem>();
            Tenants = new HashSet<Tenant>();
        }

        #endregion

        #region [ Properties ]
        public string MailBoxRecipient { get; set; }

        public string IncomingServer { get; set; }

        public string OutgoingServer { get; set; }

        public string Username { get; set; }

        public string Password { get; set; }

        public string RedirectAnomaliesSMTP { get; set; }

        public string RedirectAnomaliesUsername { get; set; }

        public string RedirectAnomaliesPassword { get; set; }

        public string RedirectAnomaliesRecipient { get; set; }

        public string RedirectStorageSMTP { get; set; }

        public string RedirectStorageUsername { get; set; }

        public string RedirectStoragePassword { get; set; }

        public string RedirectStorageRecipient { get; set; }

        public bool IsForInterop { get; set; }

        public bool? IsDestinationEnabled { get; set; }

        public bool? DeleteMailFromServer { get; set; }

        public short? ReceiveDaysCap { get; set; }

        public bool? Managed { get; set; }

        public bool? Unmanaged { get; set; }

        public int? IdConfiguration { get; set; }

        public int? IncomingServerProtocol { get; set; }

        public int? IncomingServerPort { get; set; }

        public bool? IncomingServerUseSsl { get; set; }

        public int? OutgoingServerPort { get; set; }

        public bool? OutgoingServerUseSsl { get; set; }

        public bool? IsHandleEnabled { get; set; }

        public bool? IsProtocolBox { get; set; }

        public bool? IsProtocolBoxExplicit { get; set; }

        public Guid? IdJeepServiceIncomingHost { get; set; }

        public Guid? IdJeepServiceOutgoingHost { get; set; }

        public string RulesetDefinition { get; set; }

        public InvoiceType? InvoiceType { get; set; }

        public bool LoginError { get; set; }

        #endregion

        #region[ Navigation Properties ]

        public virtual Location Location { get; set; }
        public virtual ICollection<PECMail> PECMails { get; set; }
        public virtual ICollection<OChartItem> OChartItems { get; set; }
        public virtual ICollection<Tenant> Tenants { get; set; }

        #endregion
    }
}
